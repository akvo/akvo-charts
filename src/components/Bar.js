import React from 'react';
import { useECharts } from '../hooks';
import styles from '../styles.module.css';

const getOptions = ({
  horizontal = false,
  dimensions = [],
  overrideItemStyle
}) => {
  const series = dimensions.slice(1).map((dim) => ({
    name: dim,
    type: 'bar',
    encode: {
      x: horizontal ? dim : 'category',
      y: horizontal ? 'category' : dim
    },
    ...overrideItemStyle
  }));

  return {
    series
  };
};

const Bar = ({ config, data, rawConfig }) => {
  const chartRef = useECharts({
    rawOverrides: {
      type: 'bar'
    },
    rawConfig,
    config,
    data,
    getOptions: ({ dimensions, overrideItemStyle, horizontal }) =>
      getOptions({ horizontal, dimensions, overrideItemStyle })
  });

  return (
    <div
      ref={chartRef}
      role="figure"
      className={styles.container}
    />
  );
};

export default Bar;
