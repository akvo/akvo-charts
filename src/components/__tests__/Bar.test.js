import React from 'react';
import { render, screen } from '@testing-library/react';
import Bar from '../Bar';

test('renders Bar component', () => {
  const data = [
    { label: 'A', value: 10 },
    { label: 'B', value: 20 },
    { label: 'C', value: 30 }
  ];

  const config = {
    title: 'Bar Chart Example',
    xAxisLabel: 'Categories',
    yAxisLabel: 'Values'
  };

  render(
    <Bar
      config={config}
      data={data}
    />
  );

  const chartContainer = screen.getByRole('figure');
  expect(chartContainer).toBeInTheDocument();
});
