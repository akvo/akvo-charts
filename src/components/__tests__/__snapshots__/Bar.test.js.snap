// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Bar snapshot 1`] = `
<div
  className="container"
  role="figure"
/>
`;
